core/src/wallet/stellar/synchronizers/StellarLikeBlockchainExplorerAccountSynchronizer.cpp

```diff
@@ -0,0 +1,179 @@
+        }
+
+        bool StellarLikeBlockchainExplorerAccountSynchronizer::isSynchronizing() const {
+            return _notifier != nullptr;
+        }
+
+        void StellarLikeBlockchainExplorerAccountSynchronizer::synchronizeAccount(const std::shared_ptr<StellarLikeAccount> &account) {
+            auto self = shared_from_this();
+            auto preferences = account->getInternalPreferences()->getSubPreferences("StellarLikeBlockchainExplorerAccountSynchronizer");
+            auto state = preferences
+                    ->template getObject<StellarLikeBlockchainExplorerAccountSynchronizer::SavedState>("state");
+            synchronizeAccount(account, state);
+        }
+
+        void
+        StellarLikeBlockchainExplorerAccountSynchronizer::synchronizeAccount(const std::shared_ptr<StellarLikeAccount>& account,
+                                                                             const Option<StellarLikeBlockchainExplorerAccountSynchronizer::SavedState> &state) {
+            auto address = account->getKeychain()->getAddress()->toString();
+            auto self = shared_from_this();
+            _explorer->getAccount(address).onComplete(account->getContext(), [=] (const Try<std::shared_ptr<stellar::Account>>& acc) {
+                if (acc.isFailure() && acc.getFailure().getErrorCode() == api::ErrorCode::ACCOUNT_NOT_FOUND) {
+                    self->endSynchronization();
+                } else if (acc.isFailure())
+                   self->failSynchronization(acc.getFailure());
+                else {
+                    soci::session sql(_database->getPool());
+                    account->updateAccountInfo(sql, *acc.getValue());
+                   self->synchronizeTransactions(account, state);
+                }
+            });
+        }
+
+        void StellarLikeBlockchainExplorerAccountSynchronizer::synchronizeTransactions(
+                const std::shared_ptr<StellarLikeAccount> &account,
+                const Option<StellarLikeBlockchainExplorerAccountSynchronizer::SavedState> &state) {
+            auto address = account->getKeychain()->getAddress()->toString();
+            auto transactionCursor = state.flatMap<std::string>([] (const SavedState& s) {
+                return s.transactionPagingToken.empty() ?
+                        Option<std::string>()
+                        : Option<std::string>(s.transactionPagingToken);
+            });
+
+
+            auto self = shared_from_this();
+            _explorer->getTransactions(address, transactionCursor).onComplete(account->getContext(), [=] (const Try<stellar::TransactionVector>& txs) {
+                SavedState newState = state.getValueOr(SavedState());
+                if (txs.isFailure())
+                    self->failSynchronization(txs.getFailure());
+                else {
+                    {
+                        soci::session sql(_database->getPool());
+                        soci::transaction tr(sql);
```

Comment:
I see a lot of consistency in our codebase and we should put some effort to reduce that. `tr` is basically a transaction and we are used to name it `tx`